src/controllers/match.ts

```diff
@@ -50,7 +50,53 @@ export const getInactiveMatchs = async (req: Request, res: Response) => {
     return res.status(200).json({
       statusCode: 200,
       success: 'Matchs Found',
-      data: matchs,
+      payload: matchs,
+    });
+  } catch (error: any) {
+    return res.boom.internal(error.message);
+  }
+};
+
+export const cancelMatch = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params;
+    const match = await Match.findOneAndUpdate(
+      { id, status: MatchStatus.toBePlayed },
+      { status: MatchStatus.cancelled },
+      { new: true },
+    );
+
+    if (!match) {
+      throw new Error('Match not found');
+    }
+
+    return res.status(200).json({
+      statusCode: 200,
+      success: 'Matchs Cancelled',
+      payload: match,
+    });
+  } catch (error: any) {
+    return res.boom.internal(error.message);
+  }
+};
+
+export const finishMatch = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params;
+    const match = await Match.findOneAndUpdate(
+      { id, status: MatchStatus.toBePlayed },
+      { status: MatchStatus.finished },
+      { new: true },
+    );
+
+    if (!match) {
+      throw new Error('Match not found');
+    }
```

Comment:
You need to find by `_id`, not by id. Also, we have no information about if the Match does not exist or is not in the correct status (ToBePlayed).

A better approach might be,

```suggestion
    const match = await Match.findById(id);

    if (!match) {
      throw new Error('Match not found');
    }
    
    if(math.status !== MatchStatus.toBePlayed) {
      throw new Error(`Match status is not ${MatchStatus.toBePlayed}`)
    }
    
    match.status = MatchStatus.finished
    await match.save()
```